app/models/organization.rb

```diff
@@ -14,14 +14,23 @@ class Organization < ApplicationRecord
   before_validation :downcase_slug
   before_validation :check_for_slug_change
   before_validation :evaluate_markdown
+
   before_save :update_articles
   before_save :remove_at_from_usernames
   before_save :generate_secret
+  # This callback will eventually invoke Article.update_cached_user to update the organization.name
+  # only when it has been changed, thus invoking the trigger on Article.reading_list_document
+  after_update :update_articles_cached_organization
```

Comment:
you might notice there's another callback on line 18, but it's a bit of a mistery to me how is that working. `update_articles` uses `saved_change_to_*` which [according to Rails](https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html#method-i-saved_change_to_attribute):

> This method is useful in after callbacks, to see the change in an attribute during the save that triggered the callbacks to run. It can be invoked as saved_change_to_name instead of saved_change_to_attribute("name").

I initially tried to move that as a `update_articles` as an `after_save` callback in line with what the documentation says but all hell broke loose and I started getting Elasticsearch errors. 

I didn't want to go down that rabbit hole and I think restructuring callbacks (I personally dislike them a lot but that's another story for another day) is outside the scope, so I decided to add a callback that only runs during updates.

Destroys are covered by `article_sync` which runs only on `destroy`.

Will revisit all when ES is gone